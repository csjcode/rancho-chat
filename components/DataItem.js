import React from 'react'
import { StyleSheet, Text, TouchableWithoutFeedback, View } from 'react-native'
import colors from '../constants/colors/colors'
import ProfileImage from './ProfileImage'
import { Ionicons, AntDesign } from '@expo/vector-icons'
import getColors from '../constants/colors/getColors'
const colorsTheme = getColors()

const imageSize = 40

const DataItem = (props) => {
  const { title, subTitle, image, type, isChecked, icon } = props

  const hideImage = props.hideImage && props.hideImage === true

  return (
    <TouchableWithoutFeedback onPress={props.onPress}>
      <View style={styles.container}>
        {!icon && !hideImage && <ProfileImage uri={image} size={imageSize} />}

        {icon && (
          <View style={styles.leftIconContainer}>
            <AntDesign name={icon} size={20} color={colorsTheme.blue} />
          </View>
        )}

        <View style={styles.textContainer}>
          <Text
            numberOfLines={1}
            style={{
              ...styles.title,
              ...{
                color:
                  type === 'button' ? colorsTheme.blue : colorsTheme.textColor,
              },
            }}
          >
            {title}
          </Text>

          {subTitle && (
            <Text numberOfLines={1} style={styles.subTitle}>
              {subTitle}
            </Text>
          )}
        </View>

        {type === 'checkbox' && (
          <View
            style={{
              ...styles.iconContainer,
              ...(isChecked && styles.checkedStyle),
            }}
          >
            <Ionicons name="checkmark" size={18} color="white" />
          </View>
        )}

        {type === 'link' && (
          <View>
            <Ionicons
              name="chevron-forward-outline"
              size={18}
              color={colorsTheme.grey}
            />
          </View>
        )}
      </View>
    </TouchableWithoutFeedback>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 7,
    borderBottomColor: colorsTheme.extraLightGrey,
    borderBottomWidth: 1,
    alignItems: 'center',
    minHeight: 50,
  },
  textContainer: {
    marginLeft: 14,
    flex: 1,
  },
  title: {
    fontFamily: 'medium',
    fontSize: 16,
    letterSpacing: 0.3,
  },
  subTitle: {
    fontFamily: 'regular',
    color: colorsTheme.grey,
    letterSpacing: 0.3,
  },
  iconContainer: {
    borderWidth: 1,
    borderRadius: 50,
    borderColor: colorsTheme.lightGrey,
    backgroundColor: 'white',
  },
  checkedStyle: {
    backgroundColor: colorsTheme.primary,
    borderColor: 'transparent',
  },
  leftIconContainer: {
    backgroundColor: colorsTheme.extraLightGrey,
    borderRadius: 50,
    alignItems: 'center',
    justifyContent: 'center',
    width: imageSize,
    height: imageSize,
  },
})

export default DataItem
