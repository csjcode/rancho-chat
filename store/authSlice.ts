import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface AuthState {
  token: string | null;
  userData: object | null;
  didTryAutoLogin: boolean;
}

const initialState: AuthState = {
  token: null,
  userData: null,
  didTryAutoLogin: false,
};

const authSlice = createSlice({
  name: "auth",
  initialState: {
    token: null,
    userData: null,
    didTryAutoLogin: false,
  } as AuthState,
  reducers: {
    authenticate: (
      state,
      action: PayloadAction<{ token: string; userData: object }>
    ) => {
      const { payload } = action;
      state.token = payload.token;
      state.userData = payload.userData;
      state.didTryAutoLogin = true;
    },
    setDidTryAutoLogin: (state, action) => {
      state.didTryAutoLogin = true;
    },
    logout: (state, action) => {
      state.token = null;
      state.userData = null;
      state.didTryAutoLogin = false;
    },
    updateLoggedInUserData: (
      state,
      action: PayloadAction<{ newData: object }>
    ) => {
      state.userData = { ...state.userData, ...action.payload.newData };
    },
  },
});

export const setDidTryAutoLogin = authSlice.actions.setDidTryAutoLogin;
export const authenticate = authSlice.actions.authenticate;
export const updateLoggedInUserData = authSlice.actions.updateLoggedInUserData;
export const logout = authSlice.actions.logout;
export default authSlice.reducer;
