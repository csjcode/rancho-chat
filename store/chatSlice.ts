import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface ChatsState {
  chatsData: { [key: string]: any };
}

const initialState: ChatsState = {
  chatsData: {},
};

const chatSlice = createSlice({
  name: "chats",
  initialState,
  reducers: {
    setChatsData: (state, action: PayloadAction<ChatsState>) => {
      state.chatsData = { ...action.payload.chatsData };
    },
  },
});

export const { setChatsData } = chatSlice.actions;
export default chatSlice.reducer;
