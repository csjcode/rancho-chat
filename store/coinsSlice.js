import { createSlice } from '@reduxjs/toolkit'

const coinsSlice = createSlice({
  name: 'coins',
  initialState: {
    storedCoins: {
      coins: ['solana', 'chainlink', 'stepn', 'audius', 'msol'],
    },
  },
  reducers: {
    setStoredCoins: (state, action) => {
      console.log(`action ${JSON.stringify(action)}`)
      const newCoins = action.payload
      state.storedCoins = newCoins
    },
    removeStoredCoin: (state, action) => {
      // {"type":"coins/removeStoredCoin","payload":{"coins":["solana","chainlink","stepn","audius","msol"],"removeCoin":"audius"}}
      console.log(`removeStoredCoin ${JSON.stringify(action)}`)
      const newCoins = action.payload.coins.filter(
        (item) => item !== action.payload.removeCoin,
      )
      // console.log(`action ${JSON.stringify(action)}`)
      // const newCoins = state.storedCoins.coins.pop(action.payload)
      state.storedCoins.coins = newCoins
    },
    addStoredTokenList: (state, action) => {
      console.log(`action ${JSON.stringify(action.payload)}`)
      const newCoins = Array.from(
        new Set([
          ...action.payload.coins,
          ...action.payload.addTokenListStored,
        ]),
      )
      console.log(`newCoins ${newCoins}`)
      state.storedCoins.coins = newCoins
    },
  },
})
export const setStoredCoins = coinsSlice.actions.setStoredCoins
export const removeStoredCoin = coinsSlice.actions.removeStoredCoin
export const addStoredTokenList = coinsSlice.actions.addStoredTokenList
export default coinsSlice.reducer
