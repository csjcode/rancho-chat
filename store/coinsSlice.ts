
import { createSlice } from '@reduxjs/toolkit';

interface StoredCoinsState {
  storedCoins: {
    coins: string[];
  };
}

interface SetStoredCoinsAction {
  payload: StoredCoinsState['storedCoins'];
}

interface RemoveStoredCoinAction {
  payload: {
    coins: string[];
    removeCoin: string;
  };
}

interface AddStoredTokenListAction {
  payload: {
    coins: string[];
    addTokenListStored: string[];
  };
}

const coinsSlice = createSlice({
  name: 'coins',
  initialState: {
    storedCoins: {
      coins: ['solana', 'chainlink', 'stepn', 'audius', 'msol'],
    },
  },
  reducers: {
    setStoredCoins: (state, action: SetStoredCoinsAction) => {
      console.log(`action ${JSON.stringify(action)}`);
      const { payload } = action;
      state.storedCoins = payload;
    },
    removeStoredCoin: (state, action: RemoveStoredCoinAction) => {
      console.log(`removeStoredCoin ${JSON.stringify(action)}`);
      const { payload } = action;
      const { coins, removeCoin } = payload;
      const newCoins = coins.filter((item) => item !== removeCoin);
      state.storedCoins.coins = newCoins;
    },
    addStoredTokenList: (state, action: AddStoredTokenListAction) => {
      console.log(`action ${JSON.stringify(action.payload)}`);
      const { payload } = action;
      const { coins, addTokenListStored } = payload;
      const newCoins = Array.from(
        new Set([...coins, ...addTokenListStored]),
      );
      console.log(`newCoins ${newCoins}`);
      state.storedCoins.coins = newCoins;
    },
  },
});
export const setStoredCoins = coinsSlice.actions.setStoredCoins
export const removeStoredCoin = coinsSlice.actions.removeStoredCoin
export const addStoredTokenList = coinsSlice.actions.addStoredTokenList
export default coinsSlice.reducer
