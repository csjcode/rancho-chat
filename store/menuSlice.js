import { createSlice } from '@reduxjs/toolkit'

const menuSlice = createSlice({
  name: 'menu',
  initialState: {
    storedMenu: {
      map: true,
      tricks: true,
      themeColorLight: false,
    },
  },
  reducers: {
    setStoredMenu: (state, action) => {
      // console.log(`action ${JSON.stringify(action)}`)
      const newMenu = action.payload
      state.storedMenu = newMenu
      // console.log(state.storedMenu)
    },
  },
})
export const setStoredMenu = menuSlice.actions.setStoredMenu
export default menuSlice.reducer
