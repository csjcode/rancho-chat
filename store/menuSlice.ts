import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface StoredMenu {
  map: boolean;
  tricks: boolean;
  themeColorsName: string;
}

interface MenuState {
  storedMenu: StoredMenu;
}

const initialState: MenuState = {
  storedMenu: {
    map: true,
    tricks: true,
    themeColorsName: "dark",
  },
};

const menuSlice = createSlice({
  name: "menu",
  initialState,
  reducers: {
    setStoredMenu: (state, action: PayloadAction<StoredMenu>) => {
      // console.log(`action ${JSON.stringify(action)}`)
      const newMenu = action.payload;
      state.storedMenu = newMenu;
    },
  },
});
export const setStoredMenu = menuSlice.actions.setStoredMenu;
export default menuSlice.reducer;

/*


import { createSlice } from '@reduxjs/toolkit'

const menuSlice = createSlice({
  name: 'menu',
  initialState: {
    storedMenu: {
      map: true,
      tricks: true,
      themeColorsName: 'dark',
    },
  },
  reducers: {
    setStoredMenu: (state, action) => {
      // console.log(`action ${JSON.stringify(action)}`)
      const newMenu = action.payload
      state.storedMenu = newMenu
    },
  },
})
export const setStoredMenu = menuSlice.actions.setStoredMenu
export default menuSlice.reducer

*/
