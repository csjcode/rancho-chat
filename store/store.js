import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { persistReducer, persistStore } from 'redux-persist'

import AsyncStorage from '@react-native-async-storage/async-storage'
import authSlice from './authSlice'
import chatSlice from './chatSlice'
import coinsSlice from './coinsSlice'
import menuSlice from './menuSlice'
import messagesSlice from './messagesSlice'
import notesSlice from './notesSlice'
import userSlice from './userSlice'

const persistNotesConfig = {
  key: 'notes',
  storage: AsyncStorage,
}

const persistedNotes = persistReducer(persistNotesConfig, notesSlice)

export const store = configureStore({
  reducer: {
    auth: authSlice,
    users: userSlice,
    chats: chatSlice,
    messages: messagesSlice,
    menu: menuSlice,
    coins: coinsSlice,
    notes: persistedNotes,
  },
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
})
export const persistor = persistStore(store)
