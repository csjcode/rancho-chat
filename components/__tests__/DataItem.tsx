//Test code
import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import { persistor, store } from "../../store/store";
import { Provider } from "react-redux";
import DataItem from "../DataItem";

jest.mock("redux-persist", () => {
  const real = jest.requireActual("redux-persist");
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

const image = "https://example.com/image.jpg";
const title = "Example Title";
const subTitle = "Example Subtitle";
const type = "checkbox";
const isChecked = true;
const icon = "hearto";
describe("DataItem component", () => {
  afterEach(cleanup);
  // Snapshot test
  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DataItem title={"Title"} onPress={() => alert("ok")} />
        </Provider>
      )
      .toJSON();
    // console.log(tree);
    expect(tree).toMatchSnapshot();
  });
  // Render test
  // xit("renders correctly", () => {
  //   const { getByTestId } = render(
  //     <DataItem
  //       title={title}
  //       subTitle={subTitle}
  //       image={image}
  //       type={type}
  //       isChecked={isChecked}
  //       icon={icon}
  //       onPress={() => alert("ok")}
  //     />
  //   );
  //   expect(getByTestId("data-item-title")).toHaveTextContent(title);
  //   expect(getByTestId("data-item-subtitle")).toHaveTextContent(subTitle);
  //   expect(getByTestId("data-item-icon")).toHaveStyle({ color: colors.blue });
  //   expect(getByTestId("data-item-checkbox")).toHaveStyle({
  //     backgroundColor: colors.primary,
  //   });
  //   expect(getByTestId("data-item-image")).toHaveProp("uri", image);
  // });
  // Interaction test
  // xit("triggers onPress event", () => {
  //   const onPress = jest.fn();
  //   const { getByTestId } = render(
  //     <DataItem title={title} onPress={onPress} />
  //   );
  //   fireEvent.press(getByTestId("data-item-container"));
  //   expect(onPress).toBeCalled();
  // });
});
// The tests above check the DataItem component for correct rendering, styling, and interaction. The first test checks that the component renders correctly using a snapshot test. The second and third tests check that the component renders the correct title, subtitle, image, icon, type, and isChecked prop values. The fourth test checks that the onPress event is triggered when the component is pressed.
