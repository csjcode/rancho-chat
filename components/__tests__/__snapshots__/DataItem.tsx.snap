// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataItem component matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#ededed",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "minHeight": 50,
      "paddingVertical": 7,
    }
  }
>
  <View
    onPress={[Function]}
  >
    <Image
      source={1}
      style={
        Object {
          "borderColor": "#7f8c8d",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 40,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 14,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#32d48e",
          "fontFamily": "medium",
          "fontSize": 16,
          "letterSpacing": 0.3,
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
